src/components/OrderBtn.jsx

```diff
@@ -12,8 +12,9 @@ const OrderBtn = ({ text, bgcolor, modal, handleOpen, handleCart, clearCart }) =
       fullwidth
       sx={{ textTransform: 'capitalize', cursor: 'pointer' }}
       onClick={() => {
-        modal && handleOpen();
-        handleCart && clearCart();
+        if (modal && handleOpen) handleOpen();
```

Comment:
`handleClose` would be executed regardless and override `handleOpen` because you are not returning after opening.